app/aws/s3.py

```diff
@@ -189,3 +191,33 @@ def generate_presigned_url(bucket_name: str, object_key: str, expiration: int =
         return ""
 
     return response
+
+
+def stream_to_s3(bucket_name, object_key, copy_command, cursor):
+    """
+    Stream data from PostgreSQL COPY command directly to S3.
+
+    :param bucket_name: S3 bucket name
+    :param object_key: S3 object key
+    :param copy_command: PostgreSQL COPY command
+    :param cursor: Database cursor
+    """
+    s3_client = client("s3", current_app.config["AWS_REGION"])
```

Comment:
we really should make this a class and have all these functions be under it, instead of instantiating an s3 connection for each func. (not for this PR but in the future)